### ISSUE TITLE
Scalafmt Rewrite Is Not Value Preserving

### ISSUE THREAD

**Issue by adamgfraser** [scalameta/scalafmt#1756]
This template is a guideline, not a strict requirement.
- **Version**: 0.5
- **Integration**: IntelliJ
- **Configuration**:
```
version = "2.4.2"
maxColumn = 120
align = most
continuationIndent.defnSite = 2
assumeStandardLibraryStripMargin = true
docstrings = JavaDoc
lineEndings = preserve
includeCurlyBraceInSelectChains = false
danglingParentheses = true
spaces {
  inImportCurlyBraces = true
}
optIn.annotationNewlines = true

rewrite.rules = [SortImports, RedundantBraces]

```

## Steps

Given code like this:
```scala
      childContext.onDone { _ =>
        val _ = self.withChildren(_.remove(childContext))
      }
```

When I run scalafmt like this:

```bash
scalafmt
```

## Problem

Scalafmt formats code like this:
```scala
      childContext.onDone(_ => val _ = self.withChildren(_.remove(childContext)))
```

## Expectation

I would like the formatted output to look like this:
```scala
      childContext.onDone { _ =>
        val _ = self.withChildren(_.remove(childContext))
      }
```

**Comment by kitbellew** [comment]
@adamgfraser could you explain the title of this issue, please? does it mean that the current formatting doesn't result in equivalent scala?

**Comment by adamgfraser** [comment]
@kitbellew Sure, sorry. The original code is valid code that compiles. The rewritten code does not compile. The assignment syntax is only valid within a block.

**Comment by kitbellew** [comment]
@adamgfraser ah, thank you. a similar problem (#1718) was fixed in #1715 and released in 2.4.2. when i run your test on 2.4.2, it produces
```
childContext.onDone { _ => val _ = self.withChildren(_.remove(childContext)) }
```
are you certain you are experiencing it with 2.4.2 and *not* an earlier version?

**Comment by adamgfraser** [comment]
@kitbellew Very certain. We encountered it just after merging a PR to update to 2.4.2.
